src/setup/node.rs

```diff
@@ -260,8 +277,118 @@ impl Drop for Node {
     }
 }
 
-fn get_stateful_node_path() -> io::Result<PathBuf> {
-    // TODO support multiple nodes
+fn get_stateful_node_path(node_dir: usize) -> io::Result<PathBuf> {
     let ziggurat_path = build_ripple_work_path()?;
-    Ok(ziggurat_path.join(STATEFUL_NODES_DIR).join("1"))
+    Ok(ziggurat_path
+        .join(STATEFUL_NODES_DIR)
+        .join(node_dir.to_string()))
+}
+
+#[cfg(test)]
+mod test {
+    use tempfile::TempDir;
+    use tokio::time::sleep;
+
+    use super::*;
+
+    const STATELESS_NODE_CNT: usize = 3; // Any number should work
+    const STATEFUL_NODES_CNT: usize = STATEFUL_NODES_COUNT;
+    const TOO_MANY_STATEFUL_NODES_CNT: usize = STATEFUL_NODES_CNT + 1;
+
+    const SLEEP: Duration = Duration::from_millis(10);
```

Comment:
```suggestion
    const SLEEP: Duration = Duration::from_millis(100);
```
just for extra safety

Reply:
Done